src/components/note/tag.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { ChangeEvent, KeyboardEvent, useRef, useState } from "react";
+
+import DeleteIcon from "@/assets/icons-small/delete_slate_400.svg";
+
+type Tag = {
+  id: string;
+  text: string;
+};
+
+export default function TagInput(): JSX.Element {
+  const [inputValue, setInputValue] = useState("");
+  const [tags, setTags] = useState<Tag[]>([]);
+  const tagIdCounter = useRef(0);
+
+  const getNextTagId = (): number => {
+    const nextId = tagIdCounter.current + 1;
+    tagIdCounter.current = nextId;
+    return nextId;
+  };
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === " " && inputValue.trim() !== "") {
+      e.preventDefault();
+      const newTag: Tag = {
+        id: `tag-${getNextTagId()}`,
+        text: inputValue.trim(),
+      };
+      setTags((prev) => [...prev, newTag]);
+      setInputValue("");
+    }
+  };
+
+  const handleRemoveTag = (id: string) => {
+    setTags((prev) => prev.filter((tag) => tag.id !== id));
+  };
+
+  return (
+    <div className="container-width mt-8 flex h-32 flex-nowrap overflow-hidden border-b pb-8">
+      {tags.map((tag) => (
+        <div
+          key={tag.id}
+          className="mr-4 inline-flex w-fit items-center whitespace-nowrap rounded-100 border border-dark-blue-300 bg-dark-blue-100 px-8 py-2"
+        >
+          <span className="text-sm font-normal">{tag.text}</span>
+          <button
+            className="ml-4 flex h-16 w-16 items-center justify-center"
+            type="button"
+            onClick={() => handleRemoveTag(tag.id)}
+          >
+            <DeleteIcon aria-label="태그 삭제" />
+          </button>
+        </div>
+      ))}
+      <input
+        type="text"
+        value={inputValue}
+        onChange={handleInputChange}
+        onKeyDown={handleKeyDown}
+        placeholder={tags.length > 0 ? "" : "태그를 입력하세요."}
+        aria-label="태그 입력"
+        className="container-width h-32 flex-grow border-b text-base font-normal text-slate-700 outline-none"
+      />
+    </div>
+  );
```

Comment:
태그가 있을 때, 태그와 input이 정렬이 맞으면 좀 더 보기 좋을 것 같아요! 

```suggestion
    <div className="container-width flex flex-nowrap overflow-hidden border-b ">
      {tags.map((tag) => (
        <div
          key={tag.id}
          className="mr-4 inline-flex w-fit items-center whitespace-nowrap rounded-100 border border-dark-blue-300 bg-dark-blue-100 px-8 py-2"
        >
          <span className="text-sm font-normal">{tag.text}</span>
          <button
            className="ml-4 flex h-16 w-16 items-center justify-center"
            type="button"
            onClick={() => handleRemoveTag(tag.id)}
          >
            <DeleteIcon aria-label="태그 삭제" />
          </button>
        </div>
      ))}
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        onKeyDown={handleKeyDown}
        placeholder={tags.length > 0 ? "" : "태그를 입력하세요."}
        aria-label="태그 입력"
        className="container-width h-32 flex-grow text-base font-normal text-slate-700 outline-none"
      />
    </div>
  );
```

Reply:
![image](https://github.com/user-attachments/assets/a63a62f1-1fc6-4340-ba06-3e9472f1fbd9)
유빈님이 보내주신 코드를 반영했더니, 피그마와 디자인이 달라지는 것 같은데 의도하신게 이게 맞을까요?

Reply:
상위 노트 페이지에서 데이터를 연결할거지만, const [tags, setTags] = useState<Tag[]>([]);의 tags를 사용하는 곳이 없어서(API 연결이 안되어서) 오류가 나서 부득이하게 일단 태그 동작 방식만 팀원들과 공유하려고 PR을 올리게 되었습니다. 혹시 나중에 한번에 코드를 올리는게 나을까요?

Reply:
앗 아뇨 그렇게 해달라는 건 아니었어요! state를 상위 컴포넌트로 끌어올리고 props로 내려주시는 걸로 확인했습니다 혹시나 전역 상태 쓰시는 건지 궁금했을 뿐입니다ㅎㅎ

Reply:
@typeYu 그리고 확인해보니 제가 클래스 하나를 빼먹고 올렸네요 ㅠㅠ
이렇게입니다!

![image](https://github.com/user-attachments/assets/4b353a75-b98e-4f80-8a58-3744c4d408b4)


```suggestion
    <div className="container-width flex flex-nowrap overflow-hidden border-b py-8">
      {tags.map((tag) => (
        <div
          key={tag.id}
          className="mr-4 inline-flex w-fit items-center whitespace-nowrap rounded-100 border border-dark-blue-300 bg-dark-blue-100 px-8 py-2"
        >
          <span className="text-sm font-normal">{tag.text}</span>
          <button
            className="ml-4 flex h-16 w-16 items-center justify-center"
            type="button"
            onClick={() => handleRemoveTag(tag.id)}
          >
            <DeleteIcon aria-label="태그 삭제" />
          </button>
        </div>
      ))}
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        onKeyDown={handleKeyDown}
        placeholder={tags.length > 0 ? "" : "태그를 입력하세요."}
        aria-label="태그 입력"
        className="container-width h-32 flex-grow text-base font-normal text-slate-700 outline-none"
      />
    </div>
  );
```

Reply:
![image](https://github.com/user-attachments/assets/3c44d31f-450f-40a5-8e27-389009adecf8)
아시는대로, 피그마 상 input 위에 태그가 존재하다 보니까 그렇게 구현을 한 것 같습니다.
피드백 반영하겠습니다!